File: diffapp/test/opencv_android_jnilibs_test.dart
import 'dart:io';

import 'package:flutter_test/flutter_test.dart';

void main() {
  test('Android jniLibs に OpenCV のプリビルド .so が配置されている', () {
    const base = 'android/app/src/main/jniLibs';
    final arm64 = Directory('$base/arm64-v8a');
    final v7a = Directory('$base/armeabi-v7a');

    expect(arm64.existsSync(), isTrue,
        reason: 'arm64-v8a ディレクトリが存在すること');
    expect(v7a.existsSync(), isTrue,
        reason: 'armeabi-v7a ディレクトリが存在すること');

    bool hasOpencvSo(Directory d) {
      if (!d.existsSync()) return false;
      final files = d
          .listSync()
          .whereType<File>()
          .map((f) => f.path.split(Platform.pathSeparator).last)
          .toList();
      // 一般的な OpenCV の .so 名称の一つでも含まれているか（ここでは java4 を基準にする）
      return files.any((name) =>
          name == 'libopencv_java4.so' ||
          name == 'libopencv_world.so' ||
          name.startsWith('libopencv_'));
    }

    expect(hasOpencvSo(arm64), isTrue,
        reason: 'arm64-v8a に OpenCV の .so が必要です');
    expect(hasOpencvSo(v7a), isTrue,
        reason: 'armeabi-v7a に OpenCV の .so が必要です');
  });
}

File: src/opencv_android_jnilibs_test.ts
import { describe, it, expect } from 'vitest';
import { existsSync } from 'node:fs';
import { join } from 'node:path';

describe('Android: jniLibs に OpenCV の .so が配置されている', () => {
  const base = 'diffapp/android/app/src/main/jniLibs';
  it('ディレクトリが存在する（arm64-v8a / armeabi-v7a）', () => {
    expect(existsSync(base)).toBe(true);
    expect(existsSync(join(base, 'arm64-v8a'))).toBe(true);
    expect(existsSync(join(base, 'armeabi-v7a'))).toBe(true);
  });

  it('libopencv_java4.so が両ABIに存在する', () => {
    expect(existsSync(join(base, 'arm64-v8a', 'libopencv_java4.so'))).toBe(true);
    expect(existsSync(join(base, 'armeabi-v7a', 'libopencv_java4.so'))).toBe(true);
  });
});
